src/rgw/rgw_bucket.cc

```diff
@@ -876,42 +675,47 @@ int RGWBucket::link(RGWBucketAdminOpState& op_state, std::string *err_msg)
     }
     policy.create_default(user_info.user_id, display_name);
 
-    owner = policy.get_owner();
-    r = store->set_bucket_owner(bucket_info.bucket, owner);
+    bucket_info.owner = user_info.user_id;
+
+    // ...and encode the acl
+    attrs[RGW_ATTR_ACL].clear();
+    policy.encode(attrs[RGW_ATTR_ACL]);
```

Comment:
this was missing the call to clear(), so re-encoding the policy was appending to the old one and not taking effect

Reply:
@cbodley looks good, but maybe can do it like this:
`auto& attr = atts[RGW_ATTR_ACL];
attr.clear();
policy.encode(attr);
`

to avoid a double lookup?